connectors/src/connectors/notion/lib/upsert.ts

```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+
+import { errorFromAny } from "@connectors/lib/error";
+import logger from "@connectors/logger/logger";
+import { DataSourceConfig } from "@connectors/types/data_source_config";
+
+const DUST_API_URL = process.env.DUST_API_URL;
+if (!DUST_API_URL) {
+  throw new Error("DUST_API_URL not set");
+}
+
+export async function upsertToDatasource(
+  dataSourceConfig: DataSourceConfig,
+  documentId: string,
+  documentText: string
+) {
+  const endpoint = `${DUST_API_URL}/api/v1/w/${dataSourceConfig.workspaceId}/data_sources//${dataSourceConfig.dataSourceName}/documents/${documentId}`;
```

Comment:
@fontanierh double `//` + you need to `encodeURIComponent` the name

Reply:
fixed